src/Chainweb/Pact/PactService.hs

```diff
@@ -905,14 +906,10 @@ execValidateBlock
     -> PayloadData
     -> PactServiceM cas PayloadWithOutputs
 execValidateBlock currHeader plData = do
-    -- KILLSWITCH: The logic here involving `txSilenceDates` is to be removed in
-    -- a future version of Chainweb. This rejects any non-genesis block which is
-    -- found to contain transactions.
-    --
-    now <- liftIO getCurrentTimeIntegral
+    let !now = _bct $ _blockCreationTime currHeader
     case txSilenceDates (_blockChainwebVersion currHeader) of
-        Just (start, _) | now > start && not isGenesisBlock && not payloadIsEmpty ->
-            throwM $ BlockValidationFailure "Transactions are disabled."
+        Just end | end > now && not isGenesisBlock && not payloadIsEmpty ->
+            throwM $ BlockValidationFailure "Transactions are disabled until December 5."
```

Comment:
The code would be easier to understand and maintain if the logic for the silence date validation would be moved into a dedicated validation function:

```haskell
validateSilenceDates :: MonadThrow m => BlockHeader -> PayloadData -> m ()
validateSilenceDates bh plData = case txSilenceDates (_chainwebVersion bh) of
    Just end | end > blockTime && not isGenesisBlock && not payloadIsEmpty ->
        throwM $ BlockValidationFailure "Transactions are disabled until December 5. BlockHeader: " <> encodeToText (ObjectEncoded bh)
    Nothing -> return ()
  where
    blockTime = _bct (_blockCreationTime bh)
    payloadIsEmpty = V.null $ _payloadDataTransactions plData
```

The error message should also contain the block header encoded as JSON.

This function would then just be:

```Haskell
execValidateBlock currHeader plData = do
    validateSilenceDates currHeader plData
    -- TODO: are we actually validating the output hash here?
    withBatch $ withCheckpointerRewind mb "execValidateBlock" $ \pdbenv -> do
    !result <- playOneBlock currHeader plData pdbenv
    return $! Save currHeader result
  where
    -- ...
```